src/Lean/Expr.lean

```diff
@@ -1297,7 +1297,7 @@ If `useZeta = true`, then `let`-expressions are visited. That is, it assumes
 that zeta-reduction (aka let-expansion) is going to be used.
 -/
 def isHeadBetaTarget (e : Expr) (useZeta := false) : Bool :=
-  e.getAppFn.isHeadBetaTargetFn useZeta
+  e.isApp && e.getAppFn.isHeadBetaTargetFn useZeta
```

Comment:
Necessary change to ensure termination of `if isHeadBetaTarget e then .visit e.headBeta ...` uses